### PR TITLE
Test the time to split core and ui tests

### PR DESCRIPTION
**Pull request by kmadsen**
### Description

Test parallelizing core and ui tests, because the unit test job is the circle-ci bottleneck

### Screenshots or Gifs

Reason I'm testing this one is because, after prepare-and-assemble, the unit tests take the most time. The idea is we can potentially cut that job in half and save ourselves ~10 minutes.
Screen Shot 2021-06-16 at 8 19 33 AM

After, it makes it so unit tests are no longer the bottleneck
Screen Shot 2021-06-16 at 9 38 24 AM

This is the test that keeps failing for the screenshots :) https://github.com/mapbox/mapbox-navigation-android/issues/4510

### CODE REVIEW

**Comment by kmadsen** [comment]
Will need to update the branch protection rules to require `unit-tests-core` and `unit-tests-ui` and remove `unit-tests` in order to merge this
https://github.com/mapbox/mapbox-navigation-android/settings/branch_protection_rules/4904782

File: .circleci/config.yml [hunk 1]
```diff
@@ -448,11 +451,16 @@ jobs:
           module_target: "libnavigation-core"
       - write-workspace
 
-  unit-tests:
+  unit-tests-core:
+    executor: ndk-r21e-latest-executor
+    steps:
+      - read-workspace
+      - unit-tests-ui
```
Comment (LukasPaczos) [inline .circleci/config.yml:458]:
```suggestion
      - unit-tests-core
```
:upside_down_face:

**Comment by LukasPaczos** [comment]
giphy (2)

File: .circleci/config.yml (hunk 1 above)
Comment (kmadsen) [reply .circleci/config.yml:458]: lol

**Review by LukasPaczos — APPROVED** [review]
:shipit:

**Comment by kmadsen** [comment]
Current default workflow duration is `58m 53s`. The test run completed in [47m 2s](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-android?branch=km-circleci-optimization-split-core-ui-tests). As predicted, ~10 minutes faster

https://app.circleci.com/insights/github/mapbox/mapbox-navigation-android/workflows/default/overview
Screen Shot 2021-06-16 at 8 51 14 AM

**Review by Guardiola31337 — APPROVED** [review]

**Pull request merged**